crates/krun/src/net.rs

```diff
@@ -0,0 +1,53 @@
+use std::{
+    fmt,
+    os::{
+        fd::{AsRawFd, IntoRawFd, OwnedFd},
+        unix::net::UnixStream,
+    },
+    path::Path,
+    process::Command,
+};
+
+use anyhow::{anyhow, Context, Result};
+use log::debug;
+use rustix::net::{socketpair, AddressFamily, SocketFlags, SocketType};
+use utils::env::find_in_path;
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum NetMode {
+    PASST = 0,
+    TSI,
+}
+
+pub fn connect_to_passt<P>(passt_socket_path: P) -> Result<UnixStream>
+where
+    P: AsRef<Path> + fmt::Debug,
+{
+    Ok(UnixStream::connect(passt_socket_path)?)
+}
+
+pub fn start_passt() -> Result<OwnedFd> {
+    let (parent_fd, child_fd) = socketpair(
+        AddressFamily::UNIX,
+        SocketType::STREAM,
+        SocketFlags::empty(),
```

Comment:
Please use `SocketFlags::CLOEXEC` here, call `dup` on `child_fd`, and use the new `fd` with `passt`.

When a process exits, the kernel automatically closes its sockets. The problem here is that both `parent_fd` and `child_fd` are being inherited by the `passt`. This means that, when the parent exits, its reference to `parent_fd` is closed, but there exists another reference in the child, so the socket is never closed.

Reply:
Uhh... I never realized that. Thanks for the explanation. I was looking for a way to use `SOCK_CLOEXEC` indded, but I never found where to use it.